test.scf

```diff
@@ -120,6 +120,14 @@ provide filename and viewer and format
     $ rm file.png
     $ ls
 
+check explicit direction of history
+    $ bash -c 'set -o pipefail; for i in {down,left,right,up}wards; do git-big-picture -g --history-direction=${i} | fgrep rankdir ; done'
```

Comment:
What is the purpose of `set -o pipefail` here? Is it to ensure that the fails if the pipe to `fgrep` fails?

Reply:
The idea is to make sure that the test never succeeds if git-big-picture returns an error some time after writing the rankdir line. In Bash by default in a pipe chain the exit code is that of the last command in the chain; so if the last exits with code 0 and an earlier one exits with 1, the overall result is still 0, masking that error. For comparison:

```console
# bash             -c 'false | true ; echo $?'
0

# bash -o pipefail -c 'false | true ; echo $?'
1
```

Where needed, each individual access code can be access using array `PIPESTATUS` like so:

```console
# bash -o pipefail -c 'false | true | ( exit 2 ) ; echo "<$?> [${PIPESTATUS[@]}]"'
<2> [1 0 2]
```

Reply:
PS: In Bash's own words:
```console
# help set | fgrep -A2 pipefail
              pipefail     the return value of a pipeline is the status of
                           the last command to exit with a non-zero status,
                           or zero if no command exited with a non-zero status
```